src/wayland/seat/keyboard/keymap_file.rs

```diff
@@ -0,0 +1,101 @@
+use std::{
+    ffi::{CStr, CString},
+    fs::File,
+    io::{Seek, Write},
+    os::unix::prelude::{AsRawFd, FromRawFd, RawFd},
+    path::PathBuf,
+};
+
+use nix::{
+    fcntl::{FcntlArg, SealFlag},
+    sys::memfd::MemFdCreateFlag,
+};
+use slog::error;
+
+#[derive(Debug)]
+pub struct KeymapFile {
+    sealed: Option<SealedFile>,
+    keymap: CString,
+}
+
+impl KeymapFile {
+    pub fn new(keymap: CString, log: slog::Logger) -> Self {
+        let sealed = SealedFile::new(&keymap);
+
+        if let Err(err) = sealed.as_ref() {
+            error!(log, "Error when creating sealed keymap file: {}", err);
+        }
+
+        Self {
+            sealed: sealed.ok(),
+            keymap,
+        }
+    }
+
+    pub fn with_fd<F>(&self, supports_seeled: bool, cb: F) -> Result<(), std::io::Error>
```

Comment:
```suggestion
    pub fn with_fd<F>(&self, supports_sealed: bool, cb: F) -> Result<(), std::io::Error>
```

Reply:
Yeah, this needs some followup updates in the file too :sweat_smile:

Reply:
Done :sweat_smile: